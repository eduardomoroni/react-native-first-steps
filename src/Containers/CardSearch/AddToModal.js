import React, { Component } from 'react'
import { View, StyleSheet } from 'react-native'
import PropTypes from 'prop-types'
import { cardType } from '../../Types/CardType'
import { Modal, SubmitButtonForm, DropdownInputForm } from '../Components'
import WishListService from '../../Services/WishListService'
import { reduxForm, Field, formValueSelector } from 'redux-form'
import { connect } from 'react-redux'

import { RkText } from 'react-native-ui-kitten'

export class AddToModal extends Component {
  constructor () {
    super()
    this.state = {
      isVisible: true
    }
  }

  toggleModal () {
    this.state.isVisible = !this.state.isVisible
  }

  addCardAmountToList (listName, card, amount) {
    const { wishList } = this.props
    console.log(wishList[listName], listName, amount)
    WishListService.updateCardAmount(card, 2, wishList.have)
  }

  render () {
    const {
      card,
      wishList,
      howMuchCards,
      whichTradeList
    } = this.props

    const tradeLists = ['want', 'have']
    const numberRange = ['1', '2', '3', '4']

    console.log(wishList)
    return (
      <Modal isVisible onModalHide={() => this.toggleModal()} >
        <View style={styles.modal}>
          <RkText style={styles.title}>
            {`Add to TradeList`}
          </RkText>
          <Field name='whichTradeList' component={DropdownInputForm} dropdownItems={tradeLists} selectedValue={whichTradeList} />
          <Field name='howMuchCards' component={DropdownInputForm} dropdownItems={numberRange} selectedValue={howMuchCards} />
        </View>
        <View>
          <SubmitButtonForm onPress={() => this.addCardAmountToList(whichTradeList, card, howMuchCards)} />
        </View>
      </Modal>
    )
  }
}

AddToModal.propTypes = {
  card: cardType.isRequired,
  listToAdd: PropTypes.any
}

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  title: {
    fontSize: 18,
    textAlign: 'center'
  },
  modal: {
    backgroundColor: 'white',
    height: 400,
    borderTopRightRadius: 4,
    borderTopLeftRadius: 4
  }
})

const mapStateToProps = (state) => {
  const selector = formValueSelector('AddToModalForm')

  return {
    whichTradeList: selector(state, 'whichTradeList'),
    howMuchCards: selector(state, 'howMuchCards')
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
  }
}

const AddToModalDecorated = reduxForm({form: 'AddToModalForm'})(AddToModal)
export default connect(mapStateToProps, mapDispatchToProps)(AddToModalDecorated)
