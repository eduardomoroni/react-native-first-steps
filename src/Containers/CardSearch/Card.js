/* @flow */

import React, { PureComponent } from 'react'
import { View, Text } from 'react-native'
import styles from '../../Styles/CardStyle'
import { placeholdersToSymbols } from '../../Realm/Conversion/Placeholder'

// TODO: Make this component a dumb component
// TODO: Use prop-types
type CardProps = {
  card: any,
  showCardText: boolean
}

export const getLastPrinting = (printings: any) => {
  const keys = Object.keys(printings)
  const lastPrinting = printings[keys[keys.length - 1]]
  return lastPrinting.printing
}

export class Card extends PureComponent {
  props: CardProps

  render () {
    const { card, showCardText } = this.props
    const {
      name,
      text,
      manaCost,
      type,
      power,
      toughness,
      printings
    } = card
    return (
      <View style={styles.container}>
        {renderCardNameAndMana(name, placeholdersToSymbols(manaCost))}
        {renderCardTypeAndEdition(type, getLastPrinting(printings))}
        {renderTextAndPower(text, power, toughness, showCardText)}
      </View>
    )
  }
}

const renderCardNameAndMana = (leftText, rightText) => {
  return (
    <View style={styles.lineContainer}>
      <View style={styles.leftWord}>
        <Text key='cardName' style={styles.cardNameText}>{leftText}</Text>
      </View>
      <View style={styles.rightWord}>
        <Text key='cardManaCost' style={styles.mana}>{rightText}</Text>
      </View>
    </View>
  )
}

const renderCardTypeAndEdition = (leftText, rightText) => {
  return (
    <View style={styles.lineContainer}>
      <View style={styles.leftWord}>
        <Text key='cardType' style={styles.cardTypeText}>{leftText}</Text>
      </View>
      <View style={styles.rightWord}>
        <Text key='cardEdition'>{rightText}</Text>
      </View>
    </View>
  )
}

const renderTextAndPower = (text, power, toughness, showCardText) => {
  return (
    <View style={styles.cardTextContainer}>
      <View style={styles.cardText}>
        <Text key='cardText' style={styles.text}>{showCardText ? text : ''}</Text>
      </View>
      <View style={styles.cardPower}>
        <Text key='cardPowerAndToughness' style={styles.cardPowerToughness}>{power || toughness ? `${power}/${toughness}` : ''}</Text>
      </View>
    </View>
  )
}
