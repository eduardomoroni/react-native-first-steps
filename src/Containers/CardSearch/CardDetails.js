/* @flow */

import React, { Component } from 'react'
import { View, StyleSheet } from 'react-native'
import { Metrics } from '../../Styles/Themes'
import { CardImage } from '../Components'

type CardDetailsProps = {
  card: any
}

export default class CardDetails extends Component {
  props: CardDetailsProps

  render () {
    return (
      <View style={styles.container}>
        <CardImage card={this.props.card} />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: Metrics.navBarHeight
  }
})
