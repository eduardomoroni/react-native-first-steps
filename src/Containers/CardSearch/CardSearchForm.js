/* @flow */

import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { View, Keyboard } from 'react-native'
import { reduxForm, Field, formValueSelector } from 'redux-form'
import { searchForCards, showFormModal } from '../../Redux/Actions'
import Styles from '../../Styles/CardSearchFormStyle'
import { valuesOf } from '../../Realm/RealmService'
import type { Dispatch } from 'redux'
import {
  DropdownInputForm,
  Modal,
  MultiSelect,
  ManaIconsBar,
  SubmitButtonForm,
  NumericInputForm,
  ModalToggle,
  TextInputForm
} from '../Components'

export class CardSearchForm extends Component {
  renderThreeFieldInRow (fieldOne, fieldTwo, fieldThree) {
    return (
      <View style={Styles.multipleFieldsPerLine}>
        <View style={Styles.leftField}>
          {fieldOne}
        </View>
        <View style={Styles.middleField}>
          {fieldTwo}
        </View>
        <View style={Styles.rightField}>
          {fieldThree}
        </View>
      </View>
    )
  }

  renderTwoFieldInRow (leftField, rightField) {
    return (
      <View style={Styles.multipleFieldsPerLine}>
        <View style={Styles.leftField}>
          {leftField}
        </View>
        <View style={Styles.rightField}>
          {rightField}
        </View>
      </View>
    )
  }

  renderModal () {
    let modalContent = {}
    let { visibleModal, cardSets, cardFormats, cardRarities } = this.props

    if (visibleModal === 'cardRarity') {
      modalContent = <Field name='cardRarity' component={MultiSelect} items={cardRarities} />
    } else if (visibleModal === 'cardSet') {
      modalContent = <Field name='cardSet' component={MultiSelect} items={cardSets} />
    } else if (visibleModal === 'cardFormat') {
      modalContent = <Field name='cardFormat' component={MultiSelect} items={cardFormats} />
    } else {
      return <View />
    }
    return (
      <Modal isVisible onModalHide={() => this.props.showModal('')} >
        {modalContent}
      </Modal>
    )
  }

  render () {
    const {
      handleSubmit,
      cardTypes,
      cardSubTypes,
      cardType,
      cardSubType,
      showModal,
      searchCards,
      cardRarity,
      cardSet,
      cardFormat
    } = this.props

    const formSubmit = (formValues) => {
      Keyboard.dismiss()
      console.log('Form sent: ', formValues)
      searchCards(formValues)
    }

    const numericOperators = ['', '<', '<=', '=', '>=', '=']

    return (
      <View style={Styles.container}>
        <View style={Styles.formContainer}>
          <Field name='cardName' component={TextInputForm} />
          <Field name='cardText' component={TextInputForm} />
          { this.renderTwoFieldInRow(
            <Field name='cardType' component={DropdownInputForm} dropdownItems={cardTypes} selectedValue={cardType} />,
            <Field name='cardSubType' component={DropdownInputForm} dropdownItems={cardSubTypes} selectedValue={cardSubType} />
        )}
          { this.renderThreeFieldInRow(
            <Field name='cardPower' component={NumericInputForm} dropdownItems={numericOperators} />,
            <Field name='cardToughness' component={NumericInputForm} dropdownItems={numericOperators} />,
            <Field name='cardCMC' component={NumericInputForm} dropdownItems={numericOperators} />
        )}
          <Field name='cardColors' component={ManaIconsBar} />
          {/* Secundary Options */}
          <Field name='cardFlavorText' component={TextInputForm} />
          { this.renderTwoFieldInRow(
            <Field name='cardArtist' component={TextInputForm} />,
            <Field name='cardCollectionNumber' component={TextInputForm} keyboardType={'numeric'} maxLength={3} />
        )}
          { this.renderThreeFieldInRow(
            <ModalToggle label='cardRarity' onPress={showModal} selected={cardRarity} />,
            <ModalToggle label='cardSet' onPress={showModal} selected={cardSet} />,
            <ModalToggle label='cardFormat' onPress={showModal} selected={cardFormat} />
        )}
          <Field name='cardColorsIdentity' component={ManaIconsBar} />
          {this.renderModal()}
        </View>
        <View style={Styles.containerFooter}>
          <SubmitButtonForm onPress={handleSubmit(formSubmit)} />
        </View>
      </View>
    )
  }
}

Field.propTypes = {
  dropdownItems: PropTypes.arrayOf(PropTypes.string),
  selectedValue: PropTypes.string
}

const mapStateToProps = (state) => {
  const selector = formValueSelector('CardSearchForm')

  const cardRarities = ['Common', 'Uncommon', 'Rare', 'Mythic Rare']
  const cardTypes = valuesOf('Type')
  const cardSubtypes = valuesOf('SubType')
  const printings = valuesOf('Printing')
  const formats = valuesOf('Legality')

  cardTypes.unshift('')
  cardSubtypes.unshift('')

  return {
    cardType: selector(state, 'cardType'),
    cardSubType: selector(state, 'cardSubType'),
    cardSubTypes: cardSubtypes,
    cardTypes: cardTypes,
    cardSets: printings,
    cardFormats: formats,
    cardRarity: selector(state, 'cardRarity'),
    cardSet: selector(state, 'cardSet'),
    cardFormat: selector(state, 'cardFormat'),
    visibleModal: state.cardSearch.visibleModal,
    cardRarities: cardRarities
  }
}

const mapDispatchToProps = (dispatch: Dispatch<*>) => {
  return {
    searchCards: (cardSearchForm) => dispatch(searchForCards(cardSearchForm)),
    showModal: (modal) => dispatch(showFormModal(modal))
  }
}

const cardSearchFormDecorated = reduxForm({form: 'CardSearchForm'})(CardSearchForm)
export default connect(mapStateToProps, mapDispatchToProps)(cardSearchFormDecorated)
