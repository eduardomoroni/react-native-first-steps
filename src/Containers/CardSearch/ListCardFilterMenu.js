// @flow

import React, { Component } from 'react'
import { View, BackAndroid, Switch, Text, Picker } from 'react-native'
import { reduxForm, Field, formValueSelector } from 'redux-form'
import { connect } from 'react-redux'
import { toggleShowCardText, sortCards as sortCardsActionCreator } from '../../Redux/Actions'
import I18n from 'react-native-i18n'

const renderSwitch = (fieldProps) => {
  const { input, isEnabled } = fieldProps
  const { onChange, name } = input

  return (
    <View style={styles.rowStyle}>
      <Text>{I18n.t(name)}</Text>
      <Switch style={styles.switchStyle} value={isEnabled} onValueChange={onChange} />
    </View>
  )
}

const renderSortInput = (fieldProps) => {
  const { onChange, name, value } = fieldProps.input

  const sortValues = ['cmc', 'multiverseid', 'name', 'number', 'rarity', 'type', 'power', 'toughness']

  const renderPickerItem = (value, key) => {
    return <Picker.Item label={value} value={value} key={key} />
  }

  return (
    <View style={styles.rowStyle}>
      <Text>{I18n.t(name)}</Text>
      <Picker
        selectedValue={value.field}
        onValueChange={onChange}>
        { sortValues.map(renderPickerItem) }
      </Picker>
    </View>
  )
}

class ListCardFilterMenu extends Component {
  componentDidMount () {
    BackAndroid.addEventListener('hardwareBackPress', () => {
      if (this.context.drawer.props.open) {
        this.context.drawer.toggle()
        return true
      }
      return false
    })
  }

  componentWillReceiveProps (nextProps) {
    const {
      showCardText,
      toggleCardText,
      sortBy,
      sortCards,
      sortReverseOrder
    } = nextProps

    if (this.props.showCardText !== showCardText) {
      toggleCardText({showCardText: showCardText})
    }
    if (this.props.sortBy !== sortBy || this.props.sortReverseOrder !== sortReverseOrder) {
      sortCards({field: sortBy, reversed: sortReverseOrder})
    }
  }

  render () {
    const { showCardText, sortReverseOrder } = this.props
    return (
      <View style={styles.container}>
        <Field name='showCardText' component={renderSwitch} isEnabled={showCardText} />
        <Field name='sortReverseOrder' component={renderSwitch} isEnabled={sortReverseOrder} />
        <Field name='sortBy' component={renderSortInput} />
      </View>
    )
  }
}

ListCardFilterMenu.contextTypes = {
  drawer: React.PropTypes.object
}

ListCardFilterMenu.propTypes = {
  showCardText: React.PropTypes.bool,
  sortBy: React.PropTypes.string,
  sortReverseOrder: React.PropTypes.bool,
  toggleCardText: React.PropTypes.func,
  sortCards: React.PropTypes.func
}

Field.propTypes = {
  isEnabled: React.PropTypes.bool
}

const mapStateToProps = (state) => {
  const selector = formValueSelector('CardSearchFilter')

  return {
    initialValues: {
      showCardText: state.cardSearch.showCardText,
      sortBy: state.cardSearch.sortBy.field,
      sortReverseOrder: state.cardSearch.sortBy.reversed
    },
    showCardText: selector(state, 'showCardText'),
    sortBy: selector(state, 'sortBy'),
    sortReverseOrder: selector(state, 'sortReverseOrder')
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    toggleCardText: (filter) => dispatch(toggleShowCardText(filter)),
    sortCards: (sortParams) => dispatch(sortCardsActionCreator(sortParams))
  }
}

const styles = {
  container: {
    flex: 1,
    padding: 20
  },
  switchStyle: {
    marginBottom: 10
  },
  rowStyle: {
    height: 40
  }
}

const ListCardFilterDecorated = reduxForm({form: 'CardSearchFilter'}, mapStateToProps)(ListCardFilterMenu)
export default connect(mapStateToProps, mapDispatchToProps)(ListCardFilterDecorated)
