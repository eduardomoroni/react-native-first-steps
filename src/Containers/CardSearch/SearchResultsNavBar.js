import React, { Component } from 'react'
import { formValueSelector } from 'redux-form'
import { connect } from 'react-redux'
import { navBackButton, navTitle, navButtonBar, openDrawer } from '../../Navigation/NavBarItems'
import { switchDisplayMode as switchDisplayModeAction } from '../../Redux/Actions'
import Menu from '../CardSearch/ListCardFilterMenu'
import CustomNavBar from '../../Navigation/CustomNavBar'

class SearchResultsNavBar extends Component {
  render () {
    const { showCardsAs, switchDisplayMode } = this.props

    const settingsButton = {
      buttonProps: {name: 'sliders'},
      callback: () => openDrawer(<Menu />)
    }

    const displayModeButton = {
      buttonProps: {name: showCardsAs === 'image' ? 'th-list' : 'th-large'},
      callback: () => switchDisplayMode()
    }

    return (
      <CustomNavBar
        leftRender={navBackButton()}
        middleRender={navTitle('searching_cards')}
        rightRender={navButtonBar([displayModeButton, settingsButton])} />
    )
  }
}

const mapStateToProps = (state) => {
  const selector = formValueSelector('CardSearchFilter')
  return {
    showCardsAs: selector(state, 'showCardsAs')
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    switchDisplayMode: () => dispatch(switchDisplayModeAction())
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(SearchResultsNavBar)
