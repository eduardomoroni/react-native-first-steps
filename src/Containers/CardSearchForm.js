import React from 'react'
import { reduxForm, Field, formValueSelector } from 'redux-form'
import { connect } from 'react-redux'
import { realm } from '../Config/Realm'
import { inheritanceToArray } from '../Realm/Conversion/Realm-utils'
import { Metrics } from '../Styles/Themes'
import { searchForCards } from '../Redux/Actions'
import TextInputForm from './Components/TextInputForm'
import DropdownInputForm from './Components/DropdownInputForm'
import SubmitButtonForm from './Components/SubmitButtonForm'

import {
  StyleSheet,
  View,
  Keyboard
} from 'react-native'

type CardSearchFormProps = {
  cardTypes: any,
  cardSubTypes: any,
  cardType: string,
  cardSubType: string,
  searchCards: () => void,
  handleSubmit: any
}

const selector = formValueSelector('CardSearchForm')

let CardSearchForm = (props: CardSearchFormProps) => {
  const {
    handleSubmit,
    cardTypes,
    cardSubTypes,
    searchCards
  } = props

  const {
    cardType,
    cardSubType
  } = props

  const submit = values => {
    Keyboard.dismiss()
    searchCards(values)
  }

  return (
    <View style={Styles.container}>
      <Field name='cardName' component={TextInputForm} />
      <Field name='cardText' component={TextInputForm} />
      <Field name='cardType' component={DropdownInputForm} dropdownItems={cardTypes} selectedValue={cardType} />
      <Field name='cardSubType' component={DropdownInputForm} dropdownItems={cardSubTypes} selectedValue={cardSubType} />
      <SubmitButtonForm onPress={handleSubmit(submit)} />
    </View>
  )
}

Field.propTypes = {
  dropdownItems: React.PropTypes.arrayOf(React.PropTypes.object),
  selectedValue: React.PropTypes.string
}

const Styles = StyleSheet.create({
  container: {
    paddingTop: Metrics.navBarHeight + 15
  }
})

const mapStateToProps = (state) => {
  const cardTypes = inheritanceToArray(realm.objects('Type').snapshot())
  const cardSubtypes = inheritanceToArray(realm.objects('SubType').snapshot())

  cardTypes.unshift('')
  cardSubtypes.unshift('')

  return {
    cardTypes: cardTypes,
    cardSubTypes: cardSubtypes,
    cardName: selector(state, 'cardName'),
    cardType: selector(state, 'cardType'),
    cardSubType: selector(state, 'cardSubType'),
    cardText: selector(state, 'cardText')
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    searchCards: (cardSearchForm) => dispatch(searchForCards(cardSearchForm))
  }
}

CardSearchForm = reduxForm({form: 'CardSearchForm'})(CardSearchForm)
export default connect(mapStateToProps, mapDispatchToProps)(CardSearchForm)
