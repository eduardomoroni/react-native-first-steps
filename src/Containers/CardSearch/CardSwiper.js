import React, { Component } from 'react'
import PropTypes from 'prop-types'
import CardDetails from './CardDetails'
import { Actions as NavigationActions } from 'react-native-router-flux'
import { cardType } from '../../Types/CardType'
import { getIndex } from '../../Realm/RealmService'
import Swiper from 'react-native-swiper'

class CardSwiper extends Component {
  renderCardDetails (card, key) {
    return (
      <CardDetails card={card} key={key} />
    )
  }

  updateScreenTitle (e, state) {
    const { index } = state
    const getCardName = (index) => this.children[index].props.card.name
    NavigationActions.refresh({title: `${getCardName(index)}`})
  }

  render () {
    const { cards, card } = this.props
    let { index } = this.props

    if (card !== null) {
      index = getIndex(cards, card)
    }

    return (
      <Swiper
        ref={(component) => { this.swiper = component }}
        onMomentumScrollEnd={this.updateScreenTitle}
        style={{ flex: 1 }}
        loadMinimal
        loadMinimalSize={2}
        loop={false}
        showsPagination={false}
        index={index}
        >
        {cards.map(this.renderCardDetails)}
      </Swiper>
    )
  }
}

CardSwiper.propTypes = {
  cards: PropTypes.objectOf(cardType),
  card: cardType,
  index: PropTypes.number
}

CardSwiper.defaultProps = {
  index: 0,
  card: null
}

export default CardSwiper
