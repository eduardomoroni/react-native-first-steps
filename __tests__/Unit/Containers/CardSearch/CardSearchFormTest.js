// @flow

import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import { CardSearchForm } from '../../../../src/Containers/CardSearch/CardSearchForm'
import {
  // DropdownInputForm,
  // Modal,
  // MultiSelect,
  // ManaIconsBar,
  // NumericInputForm,
  // ModalToggle,
  // TextInputForm,
  SubmitButtonForm
} from '../../../../src/Containers/Components'
// TODO: TEST THIS
const props = {
  handleSubmit: jest.fn((callback) => callback()),
  searchCards: jest.fn(),
  cardTypes: ['um', 'dois'],
  cardSubTypes: ['tres', 'quatro'],
  cardType: 'um',
  cardSubType: 'quatro',
  showModal: '', // TODO: Test all scenarios
  cardRarity: ['commom'],
  cardSet: ['AER'],
  cardFormat: ['Standard']
}

describe('<CardSearchForm />', () => {
  const wrapper = shallow(<CardSearchForm {...props} />)

  it('Should render CardSearchForm Component', () => {
    expect(wrapper.getNode()).toMatchSnapshot()
  })

  it('Should submit form on pressing button', () => {
    wrapper.find(SubmitButtonForm).simulate('click')
    expect(props.handleSubmit.mock.calls).toHaveLength(1)
    expect(props.searchCards.mock.calls).toHaveLength(1)
  })
})
