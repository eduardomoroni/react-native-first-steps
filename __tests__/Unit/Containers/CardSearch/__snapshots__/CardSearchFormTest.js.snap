// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardSearchForm /> Should render CardSearchForm Component 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "marginHorizontal": 10,
        "paddingTop": 69,
      }
    }
  >
    <Field
      component={[Function]}
      name="cardName"
    />
    <Field
      component={[Function]}
      name="cardText"
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "flex-start",
            "marginRight": 1.6666666666666667,
          }
        }
      >
        <Field
          component={[Function]}
          dropdownItems={
            Array [
              "um",
              "dois",
            ]
          }
          name="cardType"
          selectedValue="um"
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "flex-end",
            "marginLeft": 1.6666666666666667,
          }
        }
      >
        <Field
          component={[Function]}
          dropdownItems={
            Array [
              "tres",
              "quatro",
            ]
          }
          name="cardSubType"
          selectedValue="quatro"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "flex-start",
            "marginRight": 1.6666666666666667,
          }
        }
      >
        <Field
          component={[Function]}
          dropdownItems={
            Array [
              "",
              "<",
              "<=",
              "=",
              ">=",
              "=",
            ]
          }
          name="cardPower"
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
            "marginLeft": 1.6666666666666667,
          }
        }
      >
        <Field
          component={[Function]}
          dropdownItems={
            Array [
              "",
              "<",
              "<=",
              "=",
              ">=",
              "=",
            ]
          }
          name="cardToughness"
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "flex-end",
            "marginLeft": 1.6666666666666667,
          }
        }
      >
        <Field
          component={[Function]}
          dropdownItems={
            Array [
              "",
              "<",
              "<=",
              "=",
              ">=",
              "=",
            ]
          }
          name="cardCMC"
        />
      </View>
    </View>
    <Field
      component={[Function]}
      name="cardColors"
    />
    <Field
      component={[Function]}
      name="cardFlavorText"
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "flex-start",
            "marginRight": 1.6666666666666667,
          }
        }
      >
        <Field
          component={[Function]}
          name="cardArtist"
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "flex-end",
            "marginLeft": 1.6666666666666667,
          }
        }
      >
        <Field
          component={[Function]}
          keyboardType="numeric"
          maxLength={3}
          name="cardCollectionNumber"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "flex-start",
            "marginRight": 1.6666666666666667,
          }
        }
      >
        <ModalToggle
          label="cardRarity"
          onPress=""
          selected={
            Array [
              "commom",
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
            "marginLeft": 1.6666666666666667,
          }
        }
      >
        <ModalToggle
          label="cardSet"
          onPress=""
          selected={
            Array [
              "AER",
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "flex-end",
            "marginLeft": 1.6666666666666667,
          }
        }
      >
        <ModalToggle
          label="cardFormat"
          onPress=""
          selected={
            Array [
              "Standard",
            ]
          }
        />
      </View>
    </View>
    <Field
      component={[Function]}
      name="cardColorsIdentity"
    />
    <View />
  </View>
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
      }
    }
  >
    <SubmitButtonForm
      onPress={undefined}
    />
  </View>
</View>
`;
